.github/workflows/tag-maintainers.yml

```diff
@@ -0,0 +1,235 @@
+      - name: Get Nixpkgs revision from flake.lock
+        id: get-nixpkgs
+        run: |
+          echo "rev=$(jq -r '.nodes.nixpkgs.locked.rev' flake.lock)" >> "$GITHUB_OUTPUT"
+
+      # Step 4: Install Nix with the correct configuration.
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+        with:
+          nix_path: nixpkgs=https://github.com/NixOS/nixpkgs/archive/${{ steps.get-nixpkgs.outputs.rev }}.tar.gz
+          extra_nix_config: |
+            experimental-features = nix-command flakes
+
+      # Step 5: Identify which plugin files have changed in the PR.
+      - name: Get changed plugin files
+        id: changed-files
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token || secrets.GITHUB_TOKEN }}
+        run: |
+          CHANGED_FILES=$(gh pr diff ${{ github.event.pull_request.number }} --name-only || true)
+          echo "Changed files:"
+          echo "$CHANGED_FILES"
+          echo "changed_files<<EOF" >> $GITHUB_OUTPUT
+          echo "$CHANGED_FILES" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+      # Step 6: Evaluate Nix code to find maintainers for the changed files.
+      - name: Extract maintainers from changed files
+        id: extract-maintainers
+        env:
+          PR_AUTHOR: "${{ github.event.pull_request.user.login }}"
+          CHANGED_FILES: '${{ steps.changed-files.outputs.changed_files }}'
+        run: |
+          if [[ -z "$CHANGED_FILES" ]]; then
+            echo "No plugin files changed. No maintainers to tag."
+            echo "maintainers=" >> "$GITHUB_OUTPUT"
+            exit 0
+          fi
+
+          echo "Evaluating nixvim meta system for all maintainers..."
+          # Get a JSON object mapping plugin paths to maintainer data.
+          ALL_MAINTAINERS=$(nix eval --impure --expr "
+            let
+              nixvim = import ./.;
+              lib = nixvim.inputs.nixpkgs.lib.extend nixvim.lib.overlay;
+              emptyConfig = lib.nixvim.evalNixvim {
+                modules = [ { _module.check = false; } ];
+                extraSpecialArgs = { pkgs = null; };
+              };
+              inherit (emptyConfig.config.meta) maintainers;
+            in
+              maintainers
+          " --json 2>/dev/null || echo "{}")
+
+          echo "Finding maintainers for changed files..."
+          FOUND_MAINTAINERS=$(
+            echo "$CHANGED_FILES" | while IFS= read -r FILE; do
+              [[ -z "$FILE" ]] && continue
+              PLUGIN_DIR=$(dirname "$FILE")
+
+              # Use jq to find maintainers whose path key ends with the plugin directory.
+              echo "$ALL_MAINTAINERS" | jq -r --arg plugindir "$PLUGIN_DIR" '
+                to_entries[] |
+                select(.key | endswith($plugindir)) |
+                .value[] |
+                select(has("github")) |
+                .github
+              ' 2>/dev/null
+            done
+          )
```

Comment:
We evaluated all maintainers, then query the full set with jq. I think it would be simpler to directly evaluate the intersection of files and maintainers directly in nix?